src/System.Runtime.Extensions/src/System/IO/StreamReader.cs

```diff
@@ -39,7 +32,6 @@ public class StreamReader : TextReader
         private Decoder _decoder;
         private byte[] _byteBuffer;
         private char[] _charBuffer;
-        private byte[] _preamble;   // Encoding's preamble, which identifies this encoding.
```

Comment:
Aside from its use in some `Debug.Asserts` (which we shouldn't care about), it is used in two places: `Init` and `IsPreamble`, and `IsPreamble` can be called multiple times. So it's possible that a custom encoding that hasn't provided an efficient implementation of `Preamble` could result in more allocations now. I lean towards not caring about it. The upside of this change is that all the built-in encodings won't allocate now (where previously only `Encoding.UTF8` avoided the allocation). I'm guessing the built-in encodings are going to be used the vast majority of the time vs. a custom encoding. I'm open to other ideas, though.